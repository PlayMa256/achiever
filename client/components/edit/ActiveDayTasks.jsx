import React from 'react';

import PropTypes from 'prop-types';

import SelectGroup from './SelectGroup';

import {
	SPECIAL_ACTIVITY_HOLIDAY
} from '../../utils';

import strings from '../../../shared/strings';

const ActiveDayTasks = (props) => {
	const {
		selectedActivity,
		selectedPhase,
		isHoliday,
		onPhaseSelect,
		tabIndex,
		disable,
		onActivitySelect,
		projectPhasesQuery
	} = props;

	const projectPhases = projectPhasesQuery.phases || {};

	let alternativeTextForProjectPhase = projectPhases.options ? null : strings.loading;
	if (projectPhases.options && projectPhases.options.length === 1) {
		alternativeTextForProjectPhase = projectPhases.options[0].name;
	}

	const activityOptions = selectedPhase.activities.options ? selectedPhase.activities.options : [];
	let alternativeTextForActivity = projectPhases.options ? null : strings.loading;
	if (isHoliday) {
		alternativeTextForActivity = SPECIAL_ACTIVITY_HOLIDAY.name;
	}

	return (
		<div className="active-day-tasks">
			<SelectGroup
				name="projectPhase"
				label={strings.projectPhase}
				options={projectPhases.options}
				selected={selectedPhase.id}
				onChange={onPhaseSelect(projectPhases.options)}
				showTextInstead={alternativeTextForProjectPhase}
				tabIndex={tabIndex}
				disabled={disable}
			/>
			<SelectGroup
				name="activity"
				label={strings.activity}
				options={activityOptions}
				selected={selectedActivity.id}
				onChange={onActivitySelect(selectedPhase.activities.options)}
				showTextInstead={alternativeTextForActivity}
				tabIndex={tabIndex + 1}
				disabled={disable}
			/>
		</div>
	);
};

export default ActiveDayTasks;

ActiveDayTasks.propTypes = {
	disable: PropTypes.bool,
	isHoliday: PropTypes.bool,
	onPhaseSelect: PropTypes.func.isRequired,
	onActivitySelect: PropTypes.func.isRequired,
	projectPhasesQuery: PropTypes.object.isRequired,
	selectedActivity: PropTypes.object.isRequired,
	selectedPhase: PropTypes.object.isRequired,
	tabIndex: PropTypes.number
};

ActiveDayTasks.defaultProps = {
	disable: false,
	isHoliday: false,
	tabIndex: 0
};
