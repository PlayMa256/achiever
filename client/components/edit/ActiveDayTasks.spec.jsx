import React from 'react';
import { shallow } from 'enzyme';

import ActiveDayTasks from './ActiveDayTasks';

describe('ActiveDayTasks', () => {
	let onPhaseSelect;
	let onActivitySelect;
	let projectPhasesQuery;
	let selectedActivity;
	let selectedPhase;

	beforeEach(() => {
		projectPhasesQuery = {
			error: false,
			loading: true,
			phases: {
				default: 456,
				options: [
					{
						id: 456,
						name: 'Project Phase Name',
						activities: {
							default: 7,
							options: [
								{
									id: 6,
									name: 'BugFix'
								},
								{
									id: 7,
									name: 'Code/ Document Reviews'
								},
								{
									id: 9,
									name: 'Coding'
								},
								{
									id: 65,
									name: 'Absence / Compensation'
								}
							]
						}
					},
					{
						id: 445,
						name: 'Super cool project that just has bugs!',
						activities: {
							default: 6,
							options: [
								{
									id: 6,
									name: 'BugFix'
								},
								{
									id: 10,
									name: 'Test fix'
								}
							]
						}
					}
				]
			}
		};
		selectedPhase = projectPhasesQuery.phases.options[0];		// eslint-disable-line
		selectedActivity = selectedPhase.activities.options[2];		// eslint-disable-line
		onPhaseSelect = jest.fn((value) => {
			const id = parseInt(value.id, 10);
			selectedPhase = projectPhasesQuery.phases.options.find(option => option.id === id);
		});
		onActivitySelect = jest.fn((value) => {
			const id = parseInt(value, 10);
			selectedActivity = selectedPhase.activities.options.find(option => option.id === id);
		});
	});

	describe('render', () => {
		it('should render two SelectGroups components with min props', (done) => {
			const wrapper = shallow(<ActiveDayTasks
				onPhaseSelect={onPhaseSelect}
				onActivitySelect={onActivitySelect}
				projectPhasesQuery={projectPhasesQuery}
				selectedActivity={selectedActivity}
				selectedPhase={selectedPhase}
			/>);
			expect(wrapper).toMatchSnapshot();

			const wrapperHolliday = shallow(<ActiveDayTasks
				onPhaseSelect={onPhaseSelect}
				onActivitySelect={onActivitySelect}
				projectPhasesQuery={projectPhasesQuery}
				selectedActivity={selectedActivity}
				selectedPhase={selectedPhase}
				isHoliday
			/>);
			expect(wrapperHolliday).toMatchSnapshot();
			done();
		});
		it('should render the name and not a select if only option', (done) => {
			const aloneProjectPhase = {
				error: false,
				loading: false,
				phases: {
					default: 456,
					options: [
						{
							id: 456,
							name: 'Project Phase Name',
							activities: {
								default: 7,
								options: [
									{
										id: 6,
										name: 'BugFix'
									}
								]
							}
						}
					]
				}
			};
			const wrapper = shallow(<ActiveDayTasks
				onPhaseSelect={onPhaseSelect}
				onActivitySelect={onActivitySelect}
				projectPhasesQuery={aloneProjectPhase}
				selectedActivity={selectedActivity}
				selectedPhase={selectedPhase}
			/>);
			expect(wrapper.find('SelectGroup').at(0).prop('showTextInstead'))
				.toEqual('Project Phase Name');
			done();
		});
	});
	describe('onSelect', () => {
		it('should call the props selected function with the selected element', () => {
			const wrapper = shallow(<ActiveDayTasks
				onPhaseSelect={onPhaseSelect}
				onActivitySelect={onActivitySelect}
				projectPhasesQuery={projectPhasesQuery}
				selectedActivity={selectedActivity}
				selectedPhase={selectedPhase}
			/>);
			const projectSelect = wrapper.find('[name="projectPhase"]');
			projectSelect.simulate('change', { id: 445 });

			expect(onPhaseSelect).toHaveBeenCalled();
			expect(selectedPhase).toEqual(projectPhasesQuery.phases.options[1]);

			const activitySelect = wrapper.find('[name="activity"]');
			activitySelect.simulate('change', 10);
			expect(onPhaseSelect).toHaveBeenCalled();
			expect(selectedActivity).toEqual(selectedPhase.activities.options[1]);
		});
	});
});
