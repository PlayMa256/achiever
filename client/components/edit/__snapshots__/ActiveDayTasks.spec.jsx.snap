// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveDayTasks render should render two SelectGroups components with min props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveDayTasks
    disable={false}
    isHoliday={false}
    onActivitySelect={[MockFunction]}
    onPhaseSelect={[MockFunction]}
    projectPhasesQuery={
      Object {
        "error": false,
        "loading": true,
        "phases": Object {
          "default": 456,
          "options": Array [
            Object {
              "activities": Object {
                "default": 7,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 7,
                    "name": "Code/ Document Reviews",
                  },
                  Object {
                    "id": 9,
                    "name": "Coding",
                  },
                  Object {
                    "id": 65,
                    "name": "Absence / Compensation",
                  },
                ],
              },
              "id": 456,
              "name": "Project Phase Name",
            },
            Object {
              "activities": Object {
                "default": 6,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 10,
                    "name": "Test fix",
                  },
                ],
              },
              "id": 445,
              "name": "Super cool project that just has bugs!",
            },
          ],
        },
      }
    }
    selectedActivity={
      Object {
        "id": 9,
        "name": "Coding",
      }
    }
    selectedPhase={
      Object {
        "activities": Object {
          "default": 7,
          "options": Array [
            Object {
              "id": 6,
              "name": "BugFix",
            },
            Object {
              "id": 7,
              "name": "Code/ Document Reviews",
            },
            Object {
              "id": 9,
              "name": "Coding",
            },
            Object {
              "id": 65,
              "name": "Absence / Compensation",
            },
          ],
        },
        "id": 456,
        "name": "Project Phase Name",
      }
    }
    tabIndex={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SelectGroup
          disabled={false}
          label="Projeto/fase"
          name="projectPhase"
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "activities": Object {
                  "default": 7,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 7,
                      "name": "Code/ Document Reviews",
                    },
                    Object {
                      "id": 9,
                      "name": "Coding",
                    },
                    Object {
                      "id": 65,
                      "name": "Absence / Compensation",
                    },
                  ],
                },
                "id": 456,
                "name": "Project Phase Name",
              },
              Object {
                "activities": Object {
                  "default": 6,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 10,
                      "name": "Test fix",
                    },
                  ],
                },
                "id": 445,
                "name": "Super cool project that just has bugs!",
              },
            ]
          }
          selected={456}
          showTextInstead={null}
          tabIndex={0}
        />,
        <SelectGroup
          disabled={false}
          label="Atividade"
          name="activity"
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "id": 6,
                "name": "BugFix",
              },
              Object {
                "id": 7,
                "name": "Code/ Document Reviews",
              },
              Object {
                "id": 9,
                "name": "Coding",
              },
              Object {
                "id": 65,
                "name": "Absence / Compensation",
              },
            ]
          }
          selected={9}
          showTextInstead={null}
          tabIndex={1}
        />,
      ],
      "className": "active-day-tasks",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "label": "Projeto/fase",
          "name": "projectPhase",
          "onChange": [MockFunction],
          "options": Array [
            Object {
              "activities": Object {
                "default": 7,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 7,
                    "name": "Code/ Document Reviews",
                  },
                  Object {
                    "id": 9,
                    "name": "Coding",
                  },
                  Object {
                    "id": 65,
                    "name": "Absence / Compensation",
                  },
                ],
              },
              "id": 456,
              "name": "Project Phase Name",
            },
            Object {
              "activities": Object {
                "default": 6,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 10,
                    "name": "Test fix",
                  },
                ],
              },
              "id": 445,
              "name": "Super cool project that just has bugs!",
            },
          ],
          "selected": 456,
          "showTextInstead": null,
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "label": "Atividade",
          "name": "activity",
          "onChange": [MockFunction],
          "options": Array [
            Object {
              "id": 6,
              "name": "BugFix",
            },
            Object {
              "id": 7,
              "name": "Code/ Document Reviews",
            },
            Object {
              "id": 9,
              "name": "Coding",
            },
            Object {
              "id": 65,
              "name": "Absence / Compensation",
            },
          ],
          "selected": 9,
          "showTextInstead": null,
          "tabIndex": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SelectGroup
            disabled={false}
            label="Projeto/fase"
            name="projectPhase"
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "activities": Object {
                    "default": 7,
                    "options": Array [
                      Object {
                        "id": 6,
                        "name": "BugFix",
                      },
                      Object {
                        "id": 7,
                        "name": "Code/ Document Reviews",
                      },
                      Object {
                        "id": 9,
                        "name": "Coding",
                      },
                      Object {
                        "id": 65,
                        "name": "Absence / Compensation",
                      },
                    ],
                  },
                  "id": 456,
                  "name": "Project Phase Name",
                },
                Object {
                  "activities": Object {
                    "default": 6,
                    "options": Array [
                      Object {
                        "id": 6,
                        "name": "BugFix",
                      },
                      Object {
                        "id": 10,
                        "name": "Test fix",
                      },
                    ],
                  },
                  "id": 445,
                  "name": "Super cool project that just has bugs!",
                },
              ]
            }
            selected={456}
            showTextInstead={null}
            tabIndex={0}
          />,
          <SelectGroup
            disabled={false}
            label="Atividade"
            name="activity"
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "id": 6,
                  "name": "BugFix",
                },
                Object {
                  "id": 7,
                  "name": "Code/ Document Reviews",
                },
                Object {
                  "id": 9,
                  "name": "Coding",
                },
                Object {
                  "id": 65,
                  "name": "Absence / Compensation",
                },
              ]
            }
            selected={9}
            showTextInstead={null}
            tabIndex={1}
          />,
        ],
        "className": "active-day-tasks",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Projeto/fase",
            "name": "projectPhase",
            "onChange": [MockFunction],
            "options": Array [
              Object {
                "activities": Object {
                  "default": 7,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 7,
                      "name": "Code/ Document Reviews",
                    },
                    Object {
                      "id": 9,
                      "name": "Coding",
                    },
                    Object {
                      "id": 65,
                      "name": "Absence / Compensation",
                    },
                  ],
                },
                "id": 456,
                "name": "Project Phase Name",
              },
              Object {
                "activities": Object {
                  "default": 6,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 10,
                      "name": "Test fix",
                    },
                  ],
                },
                "id": 445,
                "name": "Super cool project that just has bugs!",
              },
            ],
            "selected": 456,
            "showTextInstead": null,
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Atividade",
            "name": "activity",
            "onChange": [MockFunction],
            "options": Array [
              Object {
                "id": 6,
                "name": "BugFix",
              },
              Object {
                "id": 7,
                "name": "Code/ Document Reviews",
              },
              Object {
                "id": 9,
                "name": "Coding",
              },
              Object {
                "id": 65,
                "name": "Absence / Compensation",
              },
            ],
            "selected": 9,
            "showTextInstead": null,
            "tabIndex": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ActiveDayTasks render should render two SelectGroups components with min props 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveDayTasks
    disable={false}
    isHoliday={true}
    onActivitySelect={[MockFunction]}
    onPhaseSelect={[MockFunction]}
    projectPhasesQuery={
      Object {
        "error": false,
        "loading": true,
        "phases": Object {
          "default": 456,
          "options": Array [
            Object {
              "activities": Object {
                "default": 7,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 7,
                    "name": "Code/ Document Reviews",
                  },
                  Object {
                    "id": 9,
                    "name": "Coding",
                  },
                  Object {
                    "id": 65,
                    "name": "Absence / Compensation",
                  },
                ],
              },
              "id": 456,
              "name": "Project Phase Name",
            },
            Object {
              "activities": Object {
                "default": 6,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 10,
                    "name": "Test fix",
                  },
                ],
              },
              "id": 445,
              "name": "Super cool project that just has bugs!",
            },
          ],
        },
      }
    }
    selectedActivity={
      Object {
        "id": 9,
        "name": "Coding",
      }
    }
    selectedPhase={
      Object {
        "activities": Object {
          "default": 7,
          "options": Array [
            Object {
              "id": 6,
              "name": "BugFix",
            },
            Object {
              "id": 7,
              "name": "Code/ Document Reviews",
            },
            Object {
              "id": 9,
              "name": "Coding",
            },
            Object {
              "id": 65,
              "name": "Absence / Compensation",
            },
          ],
        },
        "id": 456,
        "name": "Project Phase Name",
      }
    }
    tabIndex={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SelectGroup
          disabled={false}
          label="Projeto/fase"
          name="projectPhase"
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "activities": Object {
                  "default": 7,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 7,
                      "name": "Code/ Document Reviews",
                    },
                    Object {
                      "id": 9,
                      "name": "Coding",
                    },
                    Object {
                      "id": 65,
                      "name": "Absence / Compensation",
                    },
                  ],
                },
                "id": 456,
                "name": "Project Phase Name",
              },
              Object {
                "activities": Object {
                  "default": 6,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 10,
                      "name": "Test fix",
                    },
                  ],
                },
                "id": 445,
                "name": "Super cool project that just has bugs!",
              },
            ]
          }
          selected={456}
          showTextInstead={null}
          tabIndex={0}
        />,
        <SelectGroup
          disabled={false}
          label="Atividade"
          name="activity"
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "id": 6,
                "name": "BugFix",
              },
              Object {
                "id": 7,
                "name": "Code/ Document Reviews",
              },
              Object {
                "id": 9,
                "name": "Coding",
              },
              Object {
                "id": 65,
                "name": "Absence / Compensation",
              },
            ]
          }
          selected={9}
          showTextInstead="Holiday"
          tabIndex={1}
        />,
      ],
      "className": "active-day-tasks",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "label": "Projeto/fase",
          "name": "projectPhase",
          "onChange": [MockFunction],
          "options": Array [
            Object {
              "activities": Object {
                "default": 7,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 7,
                    "name": "Code/ Document Reviews",
                  },
                  Object {
                    "id": 9,
                    "name": "Coding",
                  },
                  Object {
                    "id": 65,
                    "name": "Absence / Compensation",
                  },
                ],
              },
              "id": 456,
              "name": "Project Phase Name",
            },
            Object {
              "activities": Object {
                "default": 6,
                "options": Array [
                  Object {
                    "id": 6,
                    "name": "BugFix",
                  },
                  Object {
                    "id": 10,
                    "name": "Test fix",
                  },
                ],
              },
              "id": 445,
              "name": "Super cool project that just has bugs!",
            },
          ],
          "selected": 456,
          "showTextInstead": null,
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "label": "Atividade",
          "name": "activity",
          "onChange": [MockFunction],
          "options": Array [
            Object {
              "id": 6,
              "name": "BugFix",
            },
            Object {
              "id": 7,
              "name": "Code/ Document Reviews",
            },
            Object {
              "id": 9,
              "name": "Coding",
            },
            Object {
              "id": 65,
              "name": "Absence / Compensation",
            },
          ],
          "selected": 9,
          "showTextInstead": "Holiday",
          "tabIndex": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SelectGroup
            disabled={false}
            label="Projeto/fase"
            name="projectPhase"
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "activities": Object {
                    "default": 7,
                    "options": Array [
                      Object {
                        "id": 6,
                        "name": "BugFix",
                      },
                      Object {
                        "id": 7,
                        "name": "Code/ Document Reviews",
                      },
                      Object {
                        "id": 9,
                        "name": "Coding",
                      },
                      Object {
                        "id": 65,
                        "name": "Absence / Compensation",
                      },
                    ],
                  },
                  "id": 456,
                  "name": "Project Phase Name",
                },
                Object {
                  "activities": Object {
                    "default": 6,
                    "options": Array [
                      Object {
                        "id": 6,
                        "name": "BugFix",
                      },
                      Object {
                        "id": 10,
                        "name": "Test fix",
                      },
                    ],
                  },
                  "id": 445,
                  "name": "Super cool project that just has bugs!",
                },
              ]
            }
            selected={456}
            showTextInstead={null}
            tabIndex={0}
          />,
          <SelectGroup
            disabled={false}
            label="Atividade"
            name="activity"
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "id": 6,
                  "name": "BugFix",
                },
                Object {
                  "id": 7,
                  "name": "Code/ Document Reviews",
                },
                Object {
                  "id": 9,
                  "name": "Coding",
                },
                Object {
                  "id": 65,
                  "name": "Absence / Compensation",
                },
              ]
            }
            selected={9}
            showTextInstead="Holiday"
            tabIndex={1}
          />,
        ],
        "className": "active-day-tasks",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Projeto/fase",
            "name": "projectPhase",
            "onChange": [MockFunction],
            "options": Array [
              Object {
                "activities": Object {
                  "default": 7,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 7,
                      "name": "Code/ Document Reviews",
                    },
                    Object {
                      "id": 9,
                      "name": "Coding",
                    },
                    Object {
                      "id": 65,
                      "name": "Absence / Compensation",
                    },
                  ],
                },
                "id": 456,
                "name": "Project Phase Name",
              },
              Object {
                "activities": Object {
                  "default": 6,
                  "options": Array [
                    Object {
                      "id": 6,
                      "name": "BugFix",
                    },
                    Object {
                      "id": 10,
                      "name": "Test fix",
                    },
                  ],
                },
                "id": 445,
                "name": "Super cool project that just has bugs!",
              },
            ],
            "selected": 456,
            "showTextInstead": null,
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Atividade",
            "name": "activity",
            "onChange": [MockFunction],
            "options": Array [
              Object {
                "id": 6,
                "name": "BugFix",
              },
              Object {
                "id": 7,
                "name": "Code/ Document Reviews",
              },
              Object {
                "id": 9,
                "name": "Coding",
              },
              Object {
                "id": 65,
                "name": "Absence / Compensation",
              },
            ],
            "selected": 9,
            "showTextInstead": "Holiday",
            "tabIndex": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
